### PR TITLE
🐛♻️ web-api: fixes adding group member by user-name and some cleanup

### PR DESCRIPTION
**Pull request by pcrespov**
## What do these changes do?

[ReDoc](https://redocly.github.io/redoc/?url=https://raw.githubusercontent.com/ITISFoundation/osparc-simcore/364096b17c1f4e48a821eab3538690aacf6c77ce/services/web/server/src/simcore_service_webserver/api/v0/openapi.yaml)

Following manual testing on the master deployment, we identified a use-case that was still not fully functional. This PR addresses the issue with a fix (including additional test coverage) and incorporates some simple (but noisy) cleanup  refactoring in the `groups` domain.

- 🐛 **Fixes `add_group_user` with `userName`**: 
   - Resolved remaining issues not addressed in https://github.com/ITISFoundation/osparc-simcore/pull/6939
   - Added the test case `test_create_organization_and_add_users` for improved validation.  
- ♻️ **Refactoring and Cleanup**:
   - Moved `GroupID` to `models_library.groups` for better modularization.
   - Unified `GroupType` for both ORM and domain models by relocating it to `common_library.groups_enum`. 
   - 🐛 Fixes migration service `Dockerfile`. Did not include new dependency with `common_library`)

## Related issue/s

- Corrects other issue not covered by https://github.com/ITISFoundation/osparc-simcore/pull/6939
- Increases test coverage and cleanup from https://github.com/ITISFoundation/osparc-simcore/pull/6917

## How to test

```
cd services/web/server
make install-dev
pytest -vv tests/unit/with_dbs/01/groups/test_groups_handlers_users.py -m acceptance_test
```

### CODE REVIEW

**Review by giancarloromeo — APPROVED** [review]

**Review by GitHK — APPROVED** [review]

**Review by odeimaiz — APPROVED** [review]

File: packages/postgres-database/requirements/prod.txt [hunk 1]
```diff
@@ -8,4 +8,5 @@
 --requirement _base.txt
 --requirement _migration.txt
 
+simcore-common-library @ ../common-library/
```
Comment (sanderegg) [inline packages/postgres-database/requirements/prod.txt:11]: what about dev.txt and ci.txt?

**Review by sanderegg — APPROVED** [review]
all good!

File: packages/postgres-database/requirements/prod.txt (hunk 1 above)
Comment (pcrespov) [reply packages/postgres-database/requirements/prod.txt:11]: True

Comment (pcrespov) [reply packages/postgres-database/requirements/prod.txt:11]: but in any case the `migration` service is always used in production even the image tagged as dev

**Pull request merged**